libs/src/services/solanaWeb3Manager/transactionHandler.js

```diff
@@ -85,8 +87,12 @@ class TransactionHandler {
     }
   }
 
-  async _locallyConfirmTransaction (instructions, recentBlockhash, logger, skipPreflight) {
-    if (!this.feePayerKeypair) {
+  async _locallyConfirmTransaction (instructions, recentBlockhash, logger, skipPreflight, feePayerOverride = null) {
+    const feePayerKeypairOverride = (feePayerOverride && this.feePayerKeypairs)
+      ? this.feePayerKeypairs.find(keypair => keypair.publicKey.toString() === feePayerOverride)
+      : null
+    const feePayerAccount = feePayerKeypairOverride || (this.feePayerKeypairs && this.feePayerKeypairs[0])
```

Comment:
is this where an invalid keypair will throw? i.e. from identity service end, if this isn't found we throw?

Reply:
it will return the null response and missing keys error message as seen a couple of lines below